h/views/accounts.py

```diff
@@ -473,7 +482,7 @@ def post_password_form(self):
         # Called by Pyramid when the change password form is submitted.
         return form.handle_form_submission(
             self.request,
-            self.forms["password"],
+            lambda: self.password_form,
```

Comment:
I've enhanced `handle_form_submission()` to optionally accept a _callable_ as the `form` argument and call it to get the form object. This is necessary because the form now changes during the request:

* When submitting the add-password form, at first (when validating the form submission) the form is an add-password form.
* But later in the request, when the server renders the form to HTML and returns it, the form must now be the change-password form since the user's account now has a password.